process/resultsParser/resultsParser_test.go

```diff
@@ -0,0 +1,159 @@
+				{
+					Nonce:         42,
+					Data:          "@6f6b@03",
+					ReturnMessage: "foobar",
+				},
+			},
+		}
+
+		outcome, err := ParseResultOutcome(transactionResult, testPubkeyConverter)
+		require.NoError(t, err)
+		require.Equal(t, Ok, outcome.ReturnCode)
+		require.Equal(t, "foobar", outcome.ReturnMessage)
+		require.Equal(t, outcome.Values, []*bytes.Buffer{bytes.NewBuffer([]byte("03"))})
+	})
+
+	t.Run("should parse contract outcome, on signal error", func(t *testing.T) {
+		t.Parallel()
+
+		transactionResult := &transaction.ApiTransactionResult{
+			Logs: &transaction.ApiLogs{
+				Address: "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
+				Events: []*transaction.Events{
+					{
+						Identifier: "signalError",
+						Topics: [][]byte{
+							[]byte("something happened"),
+						},
+						Data: []byte("@75736572206572726f72@07"),
+					},
+				},
+			},
+		}
+
+		outcome, err := ParseResultOutcome(transactionResult, testPubkeyConverter)
+		require.NoError(t, err)
+		require.Equal(t, UserError, outcome.ReturnCode)
+		require.Equal(t, outcome.Values, []*bytes.Buffer{bytes.NewBuffer([]byte("07"))})
+	})
+
+	t.Run("should parse contract outcome, on too much gas warning", func(t *testing.T) {
+		t.Parallel()
+
+		transactionResult := &transaction.ApiTransactionResult{
+			Logs: &transaction.ApiLogs{
+				Address: "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
+				Events: []*transaction.Events{
+					{
+						Identifier: "writeLog",
+						Topics: [][]byte{
+							[]byte("@too much gas provided for processing: gas provided = 596384500, gas used = 733010"),
+						},
+						Data: []byte("@6f6b"),
+					},
+				},
+			},
+		}
+
+		outcome, err := ParseResultOutcome(transactionResult, testPubkeyConverter)
+		require.NoError(t, err)
+		require.Equal(t, Ok, outcome.ReturnCode)
+		require.Equal(t, "@too much gas provided for processing: gas provided = 596384500, gas used = 733010", outcome.ReturnMessage)
+		require.Empty(t, outcome.Values)
+	})
+
+	t.Run("should parse contract outcome, on write log where first topic equals address", func(t *testing.T) {
+		t.Parallel()
+
+		transactionResult := &transaction.ApiTransactionResult{
+			Sender: "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
+			Logs: &transaction.ApiLogs{
+				Events: []*transaction.Events{
+					{
+						Identifier: "writeLog",
+						Topics: [][]byte{
+							[]byte("ZXJkMXF5dTV3dGhsZHpyOHd4NWM5dWNnOGtqYWdnMGpmczUzczhucjN6cHozaHlwZWZzZGQ4c3N5Y3I2dGg="),
+						},
+						Data: []byte("@6f6b="),
+					},
+				},
+			},
+		}
+
+		outcome, err := ParseResultOutcome(transactionResult, testPubkeyConverter)
+		require.NoError(t, err)
+		require.Equal(t, Ok, outcome.ReturnCode)
+		require.Empty(t, outcome.Values)
+	})
+}
+
+// Tested on 1st July 2024 with 10k transactions.
+func TestResultsParser_RealWorld(t *testing.T) {
+	t.Skip()
+
+	filePath := "transactions.json"
```

Comment:
We should share this file in a way or another. Any ideas? Google Cloud Storage, private or public?

Reply:
google cloud public would be good enough.